Switch from fmt external to std standard for formatting library

Dr15Jones opened the pull request:
#### PR description:

- now that all our compilers support the new <format> standard library, switch to using it.
- this uncovered cases where we had missing standard library header includes.

#### PR validation:

Code compiles.

Dr15Jones committed:
Added missing includes

Dr15Jones committed:
Move from fmt:: to std::

Can now use the standard format library with our compilers.

Dr15Jones commented:
please test

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-49086/46324

- There are other open Pull requests which might conflict with changes you have proposed:
  - File  FWCore/Integration/test/BuildFile.xml  modified in PR(s): #32632, #47504
  - File  FWCore/MessageLogger/interface/ErrorObj.h  modified in PR(s): #37245
  - File  FWCore/MessageLogger/interface/ErrorObj.icc  modified in PR(s): #37245
  - File  FWCore/MessageLogger/interface/MessageLogger.h  modified in PR(s): #37245
  - File  FWCore/MessageLogger/interface/MessageSender.h  modified in PR(s): #37245
  - File  FWCore/MessageService/test/UnitTestClient_C.cc  modified in PR(s): #37245

Dr15Jones committed:
use std::format with SoACommon

The dependence on fmt was not being forwarded properly by SCRAM
so switching to std::format solves compiling issues.

Dr15Jones commented:
please test

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-49086/46329

- There are other open Pull requests which might conflict with changes you have proposed:
  - File  DataFormats/SoATemplate/interface/SoACommon.h  modified in PR(s): #47605, #48629, #48829, #49072
  - File  FWCore/Integration/test/BuildFile.xml  modified in PR(s): #32632, #47504
  - File  FWCore/MessageLogger/interface/ErrorObj.h  modified in PR(s): #37245
  - File  FWCore/MessageLogger/interface/ErrorObj.icc  modified in PR(s): #37245
  - File  FWCore/MessageLogger/interface/MessageLogger.h  modified in PR(s): #37245
  - File  FWCore/MessageLogger/interface/MessageSender.h  modified in PR(s): #37245
  - File  FWCore/MessageService/test/UnitTestClient_C.cc  modified in PR(s): #37245

makortel commented:
@smuzaffar The `CUDADataFormats/SiPixelCluster` has a `source_only` dependence on `DataFormats/SoATemplate`
https://github.com/cms-sw/cmssw/blob/5a165f46bfb96e5c114c9a7e65af761450c2ffbe/CUDADataFormats/SiPixelCluster/BuildFile.xml#L5
that then uses `fmt`
https://github.com/cms-sw/cmssw/blob/5a165f46bfb96e5c114c9a7e65af761450c2ffbe/DataFormats/SoATemplate/BuildFile.xml#L2

I see `fmt` has a shared library, so conceptually `CUDADataFormats/SiPixelCluster` declaring only a `source_only` dependence on `DataFormats/SoATemplate` without declaring a dependence on `fmt` is wrong.

My question is that is this really what happened with scram? If `fmt` would have been a header-only library, would it have gotten used by `CUDADataFormats/SiPixelCluster` transitively?

cmsbuild commented:
-1

**Failed Tests**:   Build 
**Size**: This PR adds an extra 112KB to repository
**Summary**: https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-94d267/48530/summary.html
**COMMIT**: fe517d964078106bc8da22f97216a5ea10120190
**CMSSW**: CMSSW_16_0_X_2025-10-07-1100/el8_amd64_gcc13
**User test area**: For local testing, you can use `/cvmfs/cms-ci.cern.ch/week0/cms-sw/cmssw/49086/48530/install.sh` to create a dev area with all the needed externals and cmssw changes.


## Build

I found compilation error when building: 

<pre>/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02910/el8_amd64_gcc13/external/gcc/13.4.0-6908cfdf803923e783448096ca4f0923/bin/c++ -c -DCMS_MICRO_ARCH='x86-64-v3' -DGNU_GCC -D_GNU_SOURCE -DTBB_USE_GLIBCXX_VERSION=130400 -DTBB_SUPPRESS_DEPRECATED_MESSAGES -DTBB_PREVIEW_RESUMABLE_TASKS=1 -DTBB_PREVIEW_TASK_GROUP_EXTENSIONS=1 -DBOOST_SPIRIT_THREADSAFE -DPHOENIX_THREADSAFE -DBOOST_MATH_DISABLE_STD_FPCLASSIFY -DBOOST_UUID_RANDOM_PROVIDER_FORCE_POSIX -DBOOST_MPL_IGNORE_PARENTHESES_WARNING -DCMSSW_GIT_HASH='CMSSW_16_0_X_2025-10-07-1100' -DPROJECT_NAME='CMSSW' -DPROJECT_VERSION='CMSSW_16_0_X_2025-10-07-1100' -Isrc -Ipoison -I/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02910/el8_amd64_gcc13/cms/cmssw-patch/CMSSW_16_0_X_2025-10-07-1100/src -I/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02910/el8_amd64_gcc13/external/pcre/8.43-6d98fda3bfd074ebb583e2d6a2c75d25/include -isystem/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02910/el8_amd64_gcc13/external/boost/1.80.0-f5051ff020fe9f6f1a74162002ac1ea0/include -I/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02910/el8_amd64_gcc13/external/bz2lib/1.0.6-d113e1c6278c07eeaff5f84db9548446/include -I/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02910/el8_amd64_gcc13/external/libuuid/2.34-5ba7a8abfc0c5fecdc448cca360c25ff/include -isystem/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02910/el8_amd64_gcc13/lcg/root/6.32.13-2ba92f62034c9fcccda180513e8d0814/include -isystem/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02910/el8_amd64_gcc13/external/tbb/v2022.0.0-ede4a930df4abec602e96e3f3ad15914/include -I/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02910/el8_amd64_gcc13/external/xz/5.6.4-b9c4ffbc390ed320a5d57fd552e29a05/include -I/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02910/el8_amd64_gcc13/external/zlib/1.2.13-589f6bb51bbeba38a7adf5a10ea8a093/include -I/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02910/el8_amd64_gcc13/external/cuda/12.9.1-cff83d5f72da96ebfea8cafd87a05296/include -I/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02910/el8_amd64_gcc13/external/eigen/3bb6a48d8c171cf20b5f8e48bfb4e424fbd4f79e-95c02b8a883b2934decb8bb53ff9b486/include -I/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02910/el8_amd64_gcc13/external/eigen/3bb6a48d8c171cf20b5f8e48bfb4e424fbd4f79e-95c02b8a883b2934decb8bb53ff9b486/include/eigen3 -I/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02910/el8_amd64_gcc13/external/md5/1.0.0-26057075013e190e56dad37d35219376/include -I/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02910/el8_amd64_gcc13/external/tinyxml2/6.2.0-67924ead96ecb4e69aad321b767979a5/include -O3 -pthread -pipe -Werror=main -Werror=pointer-arith -Werror=overlength-strings -Wno-vla -Werror=overflow -std=c++20 -ftree-vectorize -Werror=array-bounds -Werror=format-contains-nul -Werror=type-limits -fvisibility-inlines-hidden -fno-math-errno --param vect-max-version-for-alias-checks=50 -Xassembler --compress-debug-sections -Wno-error=array-bounds -Warray-bounds -fuse-ld=bfd -march=x86-64-v3 -felide-constructors -fmessage-length=0 -Wall -Wno-non-template-friend -Wno-long-long -Wreturn-type -Wextra -Wpessimizing-move -Wclass-memaccess -Wno-cast-function-type -Wno-unused-but-set-parameter -Wno-ignored-qualifiers -Wno-unused-parameter -Wunused -Wparentheses -Werror=return-type -Werror=missing-braces -Werror=unused-value -Werror=unused-label -Werror=address -Werror=format -Werror=sign-compare -Werror=write-strings -Werror=delete-non-virtual-dtor -Werror=strict-aliasing -Werror=narrowing -Werror=unused-but-set-variable -Werror=reorder -Werror=unused-variable -Werror=conversion-null -Werror=return-local-addr -Wnon-virtual-dtor -Werror=switch -fdiagnostics-show-option -Wno-unused-local-typedefs -Wno-attributes -Wno-psabi -DEIGEN_DONT_PARALLELIZE -DEIGEN_MAX_ALIGN_BYTES=64 -Wno-error=unused-variable -DBOOST_DISABLE_ASSERTS -flto=auto -fipa-icf -flto-odr-type-merging -fno-fat-lto-objects -Wodr -fPIC -MMD -MF tmp/el8_amd64_gcc13/src/HeterogeneousCore/CUDAServices/plugins/HeterogeneousCoreCUDAServicesPlugins/CUDAMonitoringService.cc.d src/HeterogeneousCore/CUDAServices/plugins/CUDAMonitoringService.cc -o tmp/el8_amd64_gcc13/src/HeterogeneousCore/CUDAServices/plugins/HeterogeneousCoreCUDAServicesPlugins/CUDAMonitoringService.cc.o
>> Compiling edm plugin src/HeterogeneousCore/CUDAServices/plugins/CUDAService.cc
/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02910/el8_amd64_gcc13/external/gcc/13.4.0-6908cfdf803923e783448096ca4f0923/bin/c++ -c -DCMS_MICRO_ARCH='x86-64-v3' -DGNU_GCC -D_GNU_SOURCE -DTBB_USE_GLIBCXX_VERSION=130400 -DTBB_SUPPRESS_DEPRECATED_MESSAGES -DTBB_PREVIEW_RESUMABLE_TASKS=1 -DTBB_PREVIEW_TASK_GROUP_EXTENSIONS=1 -DBOOST_SPIRIT_THREADSAFE -DPHOENIX_THREADSAFE -DBOOST_MATH_DISABLE_STD_FPCLASSIFY -DBOOST_UUID_RANDOM_PROVIDER_FORCE_POSIX -DBOOST_MPL_IGNORE_PARENTHESES_WARNING -DCMSSW_GIT_HASH='CMSSW_16_0_X_2025-10-07-1100' -DPROJECT_NAME='CMSSW' -DPROJECT_VERSION='CMSSW_16_0_X_2025-10-07-1100' -Isrc -Ipoison -I/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02910/el8_amd64_gcc13/cms/cmssw-patch/CMSSW_16_0_X_2025-10-07-1100/src -I/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02910/el8_amd64_gcc13/external/pcre/8.43-6d98fda3bfd074ebb583e2d6a2c75d25/include -isystem/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02910/el8_amd64_gcc13/external/boost/1.80.0-f5051ff020fe9f6f1a74162002ac1ea0/include -I/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02910/el8_amd64_gcc13/external/bz2lib/1.0.6-d113e1c6278c07eeaff5f84db9548446/include -I/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02910/el8_amd64_gcc13/external/libuuid/2.34-5ba7a8abfc0c5fecdc448cca360c25ff/include -isystem/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02910/el8_amd64_gcc13/lcg/root/6.32.13-2ba92f62034c9fcccda180513e8d0814/include -isystem/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02910/el8_amd64_gcc13/external/tbb/v2022.0.0-ede4a930df4abec602e96e3f3ad15914/include -I/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02910/el8_amd64_gcc13/external/xz/5.6.4-b9c4ffbc390ed320a5d57fd552e29a05/include -I/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02910/el8_amd64_gcc13/external/zlib/1.2.13-589f6bb51bbeba38a7adf5a10ea8a093/include -I/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02910/el8_amd64_gcc13/external/cuda/12.9.1-cff83d5f72da96ebfea8cafd87a05296/include -I/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02910/el8_amd64_gcc13/external/eigen/3bb6a48d8c171cf20b5f8e48bfb4e424fbd4f79e-95c02b8a883b2934decb8bb53ff9b486/include -I/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02910/el8_amd64_gcc13/external/eigen/3bb6a48d8c171cf20b5f8e48bfb4e424fbd4f79e-95c02b8a883b2934decb8bb53ff9b486/include/eigen3 -I/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02910/el8_amd64_gcc13/external/md5/1.0.0-26057075013e190e56dad37d35219376/include -I/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02910/el8_amd64_gcc13/external/tinyxml2/6.2.0-67924ead96ecb4e69aad321b767979a5/include -O3 -pthread -pipe -Werror=main -Werror=pointer-arith -Werror=overlength-strings -Wno-vla -Werror=overflow -std=c++20 -ftree-vectorize -Werror=array-bounds -Werror=format-contains-nul -Werror=type-limits -fvisibility-inlines-hidden -fno-math-errno --param vect-max-version-for-alias-checks=50 -Xassembler --compress-debug-sections -Wno-error=array-bounds -Warray-bounds -fuse-ld=bfd -march=x86-64-v3 -felide-constructors -fmessage-length=0 -Wall -Wno-non-template-friend -Wno-long-long -Wreturn-type -Wextra -Wpessimizing-move -Wclass-memaccess -Wno-cast-function-type -Wno-unused-but-set-parameter -Wno-ignored-qualifiers -Wno-unused-parameter -Wunused -Wparentheses -Werror=return-type -Werror=missing-braces -Werror=unused-value -Werror=unused-label -Werror=address -Werror=format -Werror=sign-compare -Werror=write-strings -Werror=delete-non-virtual-dtor -Werror=strict-aliasing -Werror=narrowing -Werror=unused-but-set-variable -Werror=reorder -Werror=unused-variable -Werror=conversion-null -Werror=return-local-addr -Wnon-virtual-dtor -Werror=switch -fdiagnostics-show-option -Wno-unused-local-typedefs -Wno-attributes -Wno-psabi -DEIGEN_DONT_PARALLELIZE -DEIGEN_MAX_ALIGN_BYTES=64 -Wno-error=unused-variable -DBOOST_DISABLE_ASSERTS -flto=auto -fipa-icf -flto-odr-type-merging -fno-fat-lto-objects -Wodr -fPIC -MMD -MF tmp/el8_amd64_gcc13/src/HeterogeneousCore/CUDAServices/plugins/HeterogeneousCoreCUDAServicesPlugins/CUDAService.cc.d src/HeterogeneousCore/CUDAServices/plugins/CUDAService.cc -o tmp/el8_amd64_gcc13/src/HeterogeneousCore/CUDAServices/plugins/HeterogeneousCoreCUDAServicesPlugins/CUDAService.cc.o
>> Compiling edm plugin src/HeterogeneousCore/CUDAServices/plugins/NVProfilerService.cc
/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02910/el8_amd64_gcc13/external/gcc/13.4.0-6908cfdf803923e783448096ca4f0923/bin/c++ -c -DCMS_MICRO_ARCH='x86-64-v3' -DGNU_GCC -D_GNU_SOURCE -DTBB_USE_GLIBCXX_VERSION=130400 -DTBB_SUPPRESS_DEPRECATED_MESSAGES -DTBB_PREVIEW_RESUMABLE_TASKS=1 -DTBB_PREVIEW_TASK_GROUP_EXTENSIONS=1 -DBOOST_SPIRIT_THREADSAFE -DPHOENIX_THREADSAFE -DBOOST_MATH_DISABLE_STD_FPCLASSIFY -DBOOST_UUID_RANDOM_PROVIDER_FORCE_POSIX -DBOOST_MPL_IGNORE_PARENTHESES_WARNING -DCMSSW_GIT_HASH='CMSSW_16_0_X_2025-10-07-1100' -DPROJECT_NAME='CMSSW' -DPROJECT_VERSION='CMSSW_16_0_X_2025-10-07-1100' -Isrc -Ipoison -I/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02910/el8_amd64_gcc13/cms/cmssw-patch/CMSSW_16_0_X_2025-10-07-1100/src -I/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02910/el8_amd64_gcc13/external/pcre/8.43-6d98fda3bfd074ebb583e2d6a2c75d25/include -isystem/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02910/el8_amd64_gcc13/external/boost/1.80.0-f5051ff020fe9f6f1a74162002ac1ea0/include -I/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02910/el8_amd64_gcc13/external/bz2lib/1.0.6-d113e1c6278c07eeaff5f84db9548446/include -I/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02910/el8_amd64_gcc13/external/libuuid/2.34-5ba7a8abfc0c5fecdc448cca360c25ff/include -isystem/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02910/el8_amd64_gcc13/lcg/root/6.32.13-2ba92f62034c9fcccda180513e8d0814/include -isystem/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02910/el8_amd64_gcc13/external/tbb/v2022.0.0-ede4a930df4abec602e96e3f3ad15914/include -I/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02910/el8_amd64_gcc13/external/xz/5.6.4-b9c4ffbc390ed320a5d57fd552e29a05/include -I/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02910/el8_amd64_gcc13/external/zlib/1.2.13-589f6bb51bbeba38a7adf5a10ea8a093/include -I/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02910/el8_amd64_gcc13/external/cuda/12.9.1-cff83d5f72da96ebfea8cafd87a05296/include -I/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02910/el8_amd64_gcc13/external/eigen/3bb6a48d8c171cf20b5f8e48bfb4e424fbd4f79e-95c02b8a883b2934decb8bb53ff9b486/include -I/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02910/el8_amd64_gcc13/external/eigen/3bb6a48d8c171cf20b5f8e48bfb4e424fbd4f79e-95c02b8a883b2934decb8bb53ff9b486/include/eigen3 -I/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02910/el8_amd64_gcc13/external/md5/1.0.0-26057075013e190e56dad37d35219376/include -I/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02910/el8_amd64_gcc13/external/tinyxml2/6.2.0-67924ead96ecb4e69aad321b767979a5/include -O3 -pthread -pipe -Werror=main -Werror=pointer-arith -Werror=overlength-strings -Wno-vla -Werror=overflow -std=c++20 -ftree-vectorize -Werror=array-bounds -Werror=format-contains-nul -Werror=type-limits -fvisibility-inlines-hidden -fno-math-errno --param vect-max-version-for-alias-checks=50 -Xassembler --compress-debug-sections -Wno-error=array-bounds -Warray-bounds -fuse-ld=bfd -march=x86-64-v3 -felide-constructors -fmessage-length=0 -Wall -Wno-non-template-friend -Wno-long-long -Wreturn-type -Wextra -Wpessimizing-move -Wclass-memaccess -Wno-cast-function-type -Wno-unused-but-set-parameter -Wno-ignored-qualifiers -Wno-unused-parameter -Wunused -Wparentheses -Werror=return-type -Werror=missing-braces -Werror=unused-value -Werror=unused-label -Werror=address -Werror=format -Werror=sign-compare -Werror=write-strings -Werror=delete-non-virtual-dtor -Werror=strict-aliasing -Werror=narrowing -Werror=unused-but-set-variable -Werror=reorder -Werror=unused-variable -Werror=conversion-null -Werror=return-local-addr -Wnon-virtual-dtor -Werror=switch -fdiagnostics-show-option -Wno-unused-local-typedefs -Wno-attributes -Wno-psabi -DEIGEN_DONT_PARALLELIZE -DEIGEN_MAX_ALIGN_BYTES=64 -Wno-error=unused-variable -DBOOST_DISABLE_ASSERTS -flto=auto -fipa-icf -flto-odr-type-merging -fno-fat-lto-objects -Wodr -fPIC -MMD -MF tmp/el8_amd64_gcc13/src/HeterogeneousCore/CUDAServices/plugins/HeterogeneousCoreCUDAServicesPlugins/NVProfilerService.cc.d src/HeterogeneousCore/CUDAServices/plugins/NVProfilerService.cc -o tmp/el8_amd64_gcc13/src/HeterogeneousCore/CUDAServices/plugins/HeterogeneousCoreCUDAServicesPlugins/NVProfilerService.cc.o
src/HeterogeneousCore/CUDAServices/plugins/NVProfilerService.cc:9:10: fatal error: fmt/printf.h: No such file or directory
    9 | #include <fmt/printf.h>
      |          ^~~~~~~~~~~~~~
compilation terminated.
gmake: *** [tmp/el8_amd64_gcc13/src/HeterogeneousCore/CUDAServices/plugins/HeterogeneousCoreCUDAServicesPlugins/NVProfilerService.cc.o] Error 1
>> Building edm plugin tmp/el8_amd64_gcc13/src/HeterogeneousCore/CUDAServices/plugins/HeterogeneousCoreCUDAServicesPlugins/libHeterogeneousCoreCUDAServicesPlugins.so
</pre>

smuzaffar commented:
> I see `fmt` has a shared library, so conceptually `CUDADataFormats/SiPixelCluster` declaring only a `source_only` dependence on `DataFormats/SoATemplate` without declaring a dependence on `fmt` is wrong.
> 
> My question is that is this really what happened with scram? If `fmt` would have been a header-only library, would it have gotten used by `CUDADataFormats/SiPixelCluster` transitively?

No, `source_only` dependency just instructs scram that dependent package use only headers from its dependency package but it does not include all of its direct/indirect dependencies. Dependent package should explicitly add the external packages dependencies (e.g. in this case `fmt`) in its `BuildFile.xml`

Dr15Jones committed:
move from fmt::format to std::format

Dr15Jones committed:
Added missing dependency on fmt

Dr15Jones commented:
please test

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-49086/46339

- There are other open Pull requests which might conflict with changes you have proposed:
  - File  DataFormats/SoATemplate/interface/SoACommon.h  modified in PR(s): #47605, #48629, #48829, #49072
  - File  FWCore/Integration/test/BuildFile.xml  modified in PR(s): #32632, #47504
  - File  FWCore/MessageLogger/interface/ErrorObj.h  modified in PR(s): #37245
  - File  FWCore/MessageLogger/interface/ErrorObj.icc  modified in PR(s): #37245
  - File  FWCore/MessageLogger/interface/MessageLogger.h  modified in PR(s): #37245
  - File  FWCore/MessageLogger/interface/MessageSender.h  modified in PR(s): #37245
  - File  FWCore/MessageService/test/UnitTestClient_C.cc  modified in PR(s): #37245

cmsbuild commented:
Pull request #49086 was updated. @BenjaminRS, @Dr15Jones, @arunhep, @atpathak, @civanch, @ctarricone, @emeschi, @fwyzard, @gabrielmscampos, @jfernan2, @kpedro88, @makortel, @mandrenguyen, @mdhildreth, @nothingface0, @perrotta, @quinnanm, @rseidita, @smorovic, @smuzaffar can you please check and sign again.

Dr15Jones committed:
move from fmt::format to std::format

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-49086/46349

- There are other open Pull requests which might conflict with changes you have proposed:
  - File  DataFormats/SoATemplate/interface/SoACommon.h  modified in PR(s): #47605, #48629, #48829, #49072
  - File  EventFilter/Utilities/src/DAQSourceModelsDTH.cc  modified in PR(s): #49107
  - File  FWCore/Integration/test/BuildFile.xml  modified in PR(s): #32632, #47504, #49107
  - File  FWCore/MessageLogger/interface/ErrorObj.h  modified in PR(s): #37245
  - File  FWCore/MessageLogger/interface/ErrorObj.icc  modified in PR(s): #37245
  - File  FWCore/MessageLogger/interface/MessageLogger.h  modified in PR(s): #37245
  - File  FWCore/MessageLogger/interface/MessageSender.h  modified in PR(s): #37245
  - File  FWCore/MessageService/test/UnitTestClient_C.cc  modified in PR(s): #37245
  - File  FWCore/Services/plugins/CPU.cc  modified in PR(s): #49107
  - File  HeterogeneousCore/CUDAServices/plugins/BuildFile.xml  modified in PR(s): #49107
  - File  IOPool/Common/bin/EdmProvDump.cc  modified in PR(s): #49107

cmsbuild commented:
Pull request #49086 was updated. @BenjaminRS, @Dr15Jones, @Moanwar, @arunhep, @atpathak, @civanch, @cmsbuild, @ctarricone, @emeschi, @fwyzard, @gabrielmscampos, @jfernan2, @kpedro88, @makortel, @mandrenguyen, @mdhildreth, @nothingface0, @perrotta, @quinnanm, @rseidita, @smorovic, @smuzaffar, @srimanob, @subirsarkar can you please check and sign again.

Dr15Jones commented:
NOTE: I've only been changing those files which no longer compile after the changes made to FWCore. The reason for the failures in other code were due to the code
- not including the proper headers or
- the package not declaring its depends on `fmt` in BuildFile.xml

Dr15Jones committed:
Add missing include of <iterator>
